bundles/action/org.openhab.action.pushover/src/main/java/org/openhab/action/pushover/internal/Pushover.java

```diff
@@ -312,33 +339,64 @@ public static boolean pushover(@ParamDoc(name = "apiKey", text = "Your applicati
             }
 
             if (!StringUtils.isEmpty(sound)) {
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, sound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, sound));
             } else if (!StringUtils.isEmpty(defaultSound)) {
-                addEncodedParameter(data, MESSAGE_KEY_SOUND, defaultSound);
+                parts.add(new StringPart(MESSAGE_KEY_SOUND, defaultSound));
             }
 
             if (isValueInList(API_HIGH_PRIORITY_LIST, priority)) {
                 if (retry >= API_MIN_RETRY_SECONDS) {
-                    addEncodedParameter(data, MESSAGE_KEY_RETRY, String.valueOf(retry));
+                    parts.add(new StringPart(MESSAGE_KEY_RETRY, String.valueOf(retry)));
                 } else {
                     logger.warn("Retry value of {} is too small. Using default value of {}.", retry,
                             API_MIN_RETRY_SECONDS);
-                    addEncodedParameter(data, MESSAGE_KEY_RETRY, String.valueOf(API_MIN_RETRY_SECONDS));
+                    parts.add(new StringPart(MESSAGE_KEY_RETRY, String.valueOf(API_MIN_RETRY_SECONDS)));
                 }
 
                 if (expire <= API_MAX_EXPIRE_SECONDS) {
-                    addEncodedParameter(data, MESSAGE_KEY_EXPIRE, String.valueOf(expire));
+                    parts.add(new StringPart(MESSAGE_KEY_EXPIRE, String.valueOf(expire)));
                 } else {
                     logger.warn("Expire value of {} is too large. Using default value of {}.", expire,
                             API_MAX_EXPIRE_SECONDS);
-                    addEncodedParameter(data, MESSAGE_KEY_EXPIRE, String.valueOf(API_MAX_EXPIRE_SECONDS));
+                    parts.add(new StringPart(MESSAGE_KEY_EXPIRE, String.valueOf(API_MAX_EXPIRE_SECONDS)));
                 }
             }
+            if (attachment != null) {
+                logger.info("Push the image attachment <{}>.", attachment);
+                try {
+                    File f = new File(attachment);
+                    FilePart fp = new FilePart(MESSAGE_KEY_ATTACHMENT, f.getName(), f);
+                    fp.setContentType("image/jpeg");
```

Comment:
https://pushover.net/api#attachments
there is no clear statement about the content type
i just testet it with JPEG images

Reply:
>Sending a PNG image from internet: sendPushoverImage("Test", "https://jaxenter.de/wp-content/uploads/2017/12/openhablogo.png")
>Doesn't work. But would be a very nice feature. Can you please add it. Thanks.

this would require to first download the file from the provided location and than send it to the pushover service.
nice idea, not a big deal to implement, but i do not see it as part of this PR

PS:
i guess, you want to send a snapshot of your surveillance camera, right?
this is how i do it:
```
val camURL = "http://user:pw@camera1/Streaming/channels/1/picture"
val file = now.toString.replaceAll("[:\\/ \\+]+", "_") +".jpg"
val outputFile = "/tmp/camera/" + file
var String curl_result = executeCommandLine("curl -m 10 -o " + outputFile + " " + camURL, 5000)
sendPushoverImage(msg, outputFile);

// delete the temp file after 60 seconds
createTimer(now.plusSeconds(60)) [|
    executeCommandLine("rm " + outputFile)
]
```
works like a charm :-)

Reply:
> > Sending a PNG image from internet: sendPushoverImage("Test", "https://jaxenter.de/wp-content/uploads/2017/12/openhablogo.png")
Doesn't work. But would be a very nice feature. Can you please add it. Thanks.

> this would require to first download the file from the provided location and than send it to the pushover service.
nice idea, not a big deal to implement, but i do not see it as part of this PR

@andre77 are you still onto this or shall we open an issue for it?

Reply:
i do not plan to implement it to be honest, since its not clear to me, that this functionality should be part of the pushover binding.
there is already a solution, for example using executeCommandLine and curl (see my previous comment). why not just do it like this?

Reply:
Your example is fine. No worries. I can work with that. But from my point of view `curl` is not an easy-to-use tool for a basic user. And the questions in the community about using an URL directly show that the people want to have such a feature directly in the binding: 

https://community.openhab.org/t/pushover-image-support/39201/53

I filed an issue for it (see #5559)